benchmarks/count_neighbors.jl

```diff
@@ -11,6 +11,15 @@ Due to the minimal computational cost, differences between neighborhood search
 implementations are highlighted. On the other hand, this is the least realistic benchmark.
 
 For a computationally heavier benchmark, see [`benchmark_n_body`](@ref).
+
+# Example
+```julia
+    nhs_trivial = TrivialNeighborhoodSearch .......
+
+    ...
+
+    nhs_grid = GridNeighborhoodSearch ....
+```
```

Comment:
I didn't add an example because this benchmark is mostly to be used with `plot_benchmarks`.